src/core/ddsi/src/ddsi_security_omg.c

```diff
@@ -686,7 +695,112 @@ static void deinit_plugin_suite_config (dds_security_plugin_suite_config *suite_
   deinit_plugin_config (&suite_config->cryptography);
 }
 
-dds_return_t q_omg_security_load (dds_security_context *sc, const dds_qos_t *qos)
+static DDS_Security_boolean on_revoke_permissions_cb(const dds_security_access_control *plugin, const DDS_Security_PermissionsHandle handle)
+{
+  struct ddsi_domaingv *gv = plugin->gv;
```

Comment:
I think this is not a problem and will also work when using multiple domains (which have their own `gv` and own `gv->security_context`), because the plugin context (e.g. `gv->security_context->authentication_context`) is allocated in the plugin's init functions for each domain.